### ISSUE TITLE
Use `bitflags` for `oxc_regular_expression::ast::Flags`

### ISSUE THREAD

**Issue by rzvxa** [oxc-project/oxc#5562]
Currently our `oxc_ast::ast::literal::RegExpFlags` is defined using the `bitflags` macro.

https://github.com/oxc-project/oxc/blob/af69393a1c72f58606e01875d69f0bad85d646e7/crates/oxc_ast/src/ast/literal.rs#L156-L197

While the `oxc_regular_expression::ast::Flags` is defined as such:

https://github.com/oxc-project/oxc/blob/af69393a1c72f58606e01875d69f0bad85d646e7/crates/oxc_regular_expression/src/ast.rs#L26-L40

I suggest we refactor `Flags` to use `bitflags` and maybe rename it to `RegularExpressionFlags`. Ideally, we should remove the `RegExpFlags` and share this flag and its parser(probably through `RegularExpressionFlags::try_from("giv")`) throughout the project.

We can still keep a `FlagsOption` that implements `Into<RegularExpressionFlags>` if we want to allow users to use a struct to pass in flags to the pattern parser.

cc @leaysgur @overlookmotel

**Comment by rzvxa** [comment]
In regards to the `span` field on the `Flags` if we find it absolutely necessary to have this field we can extract the flags to a bitflag and define it as such:


```rust
 #[ast] 
 #[derive(Debug, Clone)] 
 #[generate_derive(CloneIn, ContentEq, ContentHash)] 
 #[cfg_attr(feature = "serialize", derive(Serialize, Tsify))] 
 pub struct Flags { 
     pub span: Span, 
     pub flags: RegularExpressionFlags
 } 
```

However, I think this span is redundant since we can easily obtain it at the `RegularExpression` level which is the only place we really would access this flag normally.

**Comment by leaysgur** [comment]
> I suggest we refactor Flags to use bitflags and maybe rename it to RegularExpressionFlags.

I agree with this.

> In regards to the span field on the Flags
> However, I think this span is redundant 

From the perspective of `oxc_parser`, I think this is true.

But from the perspective of the independent `oxc_regular_expression` crate, it may be necessary...?

**Comment by rzvxa** [comment]
> But from the perspective of the independent oxc_regular_expression crate, it may be necessary...?

In what context do we need this information? Are we looking also to produce an `oxc_regular_expression::ast::RegularExpression` for regex constructor calls?

How I look at it is that a flag always contains valid values and depending on the number of flags that are true we can calculate its span from a starting point, So in the `RegularExpression` type we can calculate it like this:

```rust
impl<'a> RegularExpression<'a> {
    pub fn flags_span(&self) -> Span {
        Span::new(self.pattern.span.end + 1 /* skip `/` in the middle */, self.span.end)
    }
}
```

And we can also have this method on the `RegularExpressionFlags`:

```rust
impl RegularExpressionFlags {
    pub fn active_flags(&self) -> u8 {
        return number_of_active_flags
    }
}
```

Where do users care for flags without also having access to their regex literal?

**Comment by leaysgur** [comment]
> Are we looking also to produce an oxc_regular_expression::ast::RegularExpression for regex constructor calls?

Yes, I was thinking in case of that. However, the position of the flag text also should be known, and it seems OK for now. 😅 (sorry)

**Comment by leaysgur** [comment]
Closing, `oxc_regular_expression::ast::Flags` is removed in https://github.com/oxc-project/oxc/pull/6262
